asf_core_data/pipeline/data_joining/merge_install_dates.py

```diff
@@ -23,118 +24,74 @@ def get_mcs_install_dates(epc_df, additional_features=False):
         ],
+        dtype={"UPRN": "str"},
     )
 
-    mcs_data.fillna({"address_1": "", "address_2": "", "address_3": ""}, inplace=True)
-
     # Rename columns
     mcs_data.rename(
-        columns={
-            "date": "HP_INSTALL_DATE",
-            "tech_type": "Type of HP",
-            "compressed_epc_address": "compressed_epc_address",
-            "address_1": "MCS address 1",
-            "address_2": "MCS address 2",
-            "address_3": "MCS address 3",
-            "postcode": "MCS postcode",
-        },
+        columns={"commission_date": "HP_INSTALL_DATE"},
         inplace=True,
     )
 
-    # Get original EPC address from MCS/EPC match
-    mcs_data = mcs_data.loc[~mcs_data["compressed_epc_address"].isna()]
-    mcs_data["MCS_ADDRESS"] = (
-        mcs_data["MCS address 1"]
-        + " "
-        + mcs_data["MCS address 2"]
-        + " "
-        + mcs_data["MCS address 3"]
-        + " "
-        + mcs_data["MCS postcode"]
-    )
-
-    mcs_data["compressed_epc_address"] = (
-        mcs_data["compressed_epc_address"]
-        .str.strip()
-        .str.lower()
-        .replace(r"\s+", "", regex=True)
-    )
-
     # Get the MCS install dates
     mcs_data["HP_INSTALL_DATE"] = pd.to_datetime(
         mcs_data["HP_INSTALL_DATE"]
         .str.strip()
         .str.lower()
+        .replace(r"\s.*", "", regex=True)
         .replace(r"-", "", regex=True),
         format="%Y%m%d",
     )
 
-    # no Nans or "" in compressed_epc_address
     mcs_data = mcs_data.sort_values("HP_INSTALL_DATE", ascending=True).drop_duplicates(
-        subset=["compressed_epc_address"], keep="first"
+        subset=["UPRN"], keep="first"
     )
 
     # Create a date dict from MCS data and apply to EPC data
     # If no install date is found for address, it assigns NaN
-    date_dict = mcs_data.set_index("compressed_epc_address").to_dict()[
-        "HP_INSTALL_DATE"
-    ]
-
-    original_address_dict = mcs_data.set_index("compressed_epc_address").to_dict()[
-        "MCS_ADDRESS"
-    ]
+    date_dict = mcs_data.set_index("UPRN").to_dict()["HP_INSTALL_DATE"]
 
-    epc_df["HP_INSTALL_DATE"] = epc_df["original_address"].map(date_dict)
-    epc_df["MCS address"] = epc_df["original_address"].map(original_address_dict)
+    epc_df["HP_INSTALL_DATE"] = epc_df["UPRN"].map(date_dict)
 
     if additional_features:
-        for feat in [
-            "version",
-            "alt_type",
-            "installation_type",
-            "# records",
-        ]:
+        for feat in ["tech_type", "cluster", "installer_name"]:
 
-            epc_df[feat] = epc_df["original_address"].map(
-                mcs_data.set_index("compressed_epc_address").to_dict()[feat]
+            print(mcs_data.columns)
+
+            epc_df[feat] = epc_df["UPRN"].map(
+                mcs_data.set_index("UPRN").to_dict()[feat]
             )
 
     return epc_df
 
 
-def manage_hp_install_dates(df, identifier="UPRN", verbose=False):
+def manage_hp_install_dates(
```

Comment:
missing a couple of the arguments in the docstring :)

Reply:
Fixed, also renamed a few things.